enumerable.rb

```diff
@@ -0,0 +1,350 @@
+        yield self[i]
+        i += 1
+      end
+
+      for element in self
+        if yield element
+          j += 1
+        end
+      end
+      j > 0 and j <= self.size ? true : false
+    end
+  end
+
+  def my_none?(word = nil)
+    unless block_given?
+      i = 0
+      j = 0
+      while i < self.size
+        self[i]
+        i += 1
+      end
+      if self.size == 0
+        true
+      elsif word.class == Regexp
+
+        for element in self
+          if element.match(word)
+            j += 1
+          end
+        end
+        j == 0 ? true : false
+
+      elsif word != nil
+        if word == Numeric
+          for element in self
+            if element.class <= word
+              j += 1
+            end
+          end
+
+          j != 0 ? false : true
+        else
+          i = 0
+          k = 0
+          while i < self.size
+            if self[i].class == word
+              k += 1
+            end
+            i += 1
+          end
+          k == 0 ? true : false
+      end
+
+      else
+        j = 0
+        self.my_each do |e|
+          if e == true
+            j += 1
+          end
+        end
+        j == 0 ? true : false
+      end
+    else
+      i = 0
+      while i < self.size
+        yield self[i]
+        i += 1
+      end
+      j = 0
+      self.my_each do |element|
+        if yield element
+          j += 1
+        end
+      end
+      j == 0 ? true : false
+
+    end
+  end
+
+  def my_count(val = nil)
+    unless block_given?
+
+      if val.class != NilClass
+        i = 0
+
+        for el in self do
+          if el == val
+            i += 1
+          end
+        end
+        i
+      else
+        self.my_each do |count|
+          count += 1
+        end
+        count
+      end
+    else
+      j = 0
+      self.my_each do |element|
+        if yield element
+          j += 1
+        end
+      end
+      j
+    end
+  end
+
+  def my_map
+    unless block_given?
+      to_enum(__method__)
+    else
+      arr = []
+      for element in self do
+        current = yield element
+        arr << current
+      end
+      arr
+    end
+  end
+
+  def my_inject(*param)
+    
+    arr = []
+    i = 0
+    for element in self do
+      arr << element
+    end
+
+    unless block_given?
+
+      if param.length == 1
+        acc = param[0]
+        result = arr[0]
+        arr[1..-1].my_each do |el|
+          result = result.send(acc, el)
+        end
+      end
+      if param.length == 2
+        result = param[0]
+        acc = param[1]
+        arr.my_each do |el|
+          result = result.send(acc, el)
+        end
+      end
+    else
+      result = yield(arr[0], arr[1])
+
+      arr.my_each do |e|
+        if e.class == String
+          result = yield(result, e)
+        end
+
+        if i > 1
+          result = yield(result, e)
+        end
+        i += 1
+      end
+      if acc.class <= Numeric
+        result = result * acc
+      end
+
+  end
+    result
+  end
```

Comment:
#### my_inject 

- [ ] combines all elements of enum by applying a binary operation, specified by a block

Given
```
    operation = proc { |sum, n| sum + n } 
```
Expected result

>       (1..5).my_inject(4) { |prod, n| prod * n } should be equal (1..5).inject(4) { |prod, n| prod * n } 480